packages/cli/src/commands/gas-report.ts

```diff
@@ -0,0 +1,219 @@
+  gasReport = gasReport.map((entry) => {
+    const prevEntry = compareGasReport.find((e) => e.name === entry.name && e.functionCall === entry.functionCall);
+    return { ...entry, prevGasUsed: prevEntry?.gasUsed };
+  });
+
+  // Print gas report
+  printGasReport(gasReport, compare);
+
+  // Save gas report to file if requested
+  if (save) saveGasReport(gasReport, save);
+
+  process.exit(0);
+};
+
+async function runGasReport(path: string): Promise<GasReport> {
+  if (!path.endsWith(".t.sol")) {
+    console.log("Skipping gas report for", chalk.bold(path), "(not a test file)");
+    return [];
+  }
+  console.log("Running gas report for", chalk.bold(path));
+  const gasReport: GasReport = [];
+
+  // Parse the given test file, and add gas reporting wherever requested by a `// !gasreport` comment
+  const fileContents = readFileSync(path, "utf8");
+  let newFile = fileContents;
+
+  // Use a regex to find first line of each function
+  const functionRegex = new RegExp(/function (.*){/g);
+  // Insert a line to declare the _gasreport variable at the start of each function
+  let functionMatch;
+  while ((functionMatch = functionRegex.exec(fileContents)) !== null) {
+    const functionSignature = functionMatch[0];
+    newFile = newFile.replace(functionSignature, `${functionSignature}\nuint256 _gasreport;`);
+  }
+
+  // A gasreport comment has a name (written after the comment) and a function call (written on the next line)
+  // Create a regex to extract both the name and the function call
+  const regex = new RegExp(/\/\/ !gasreport (.*)\n(.*)/g);
+
+  // Apply the regex and loop through the matches,
+  // and create a new file with the gasreport comments replaced by the gas report
+  let match;
+  let i = 0;
+  while ((match = regex.exec(fileContents)) !== null) {
+    const name = match[1];
+    const functionCall = match[2].trim();
+
+    newFile = newFile.replace(
+      match[0],
+      `
+_gasreport = gasleft();
+${functionCall}
+_gasreport = _gasreport - gasleft();
+console.log("GAS REPORT: ${name} [${functionCall.replaceAll('"', '\\"')}]:", _gasreport);`
+    );
+
+    i++;
+  }
+
+  // Remove all occurrences of `pure` with `view`
+  newFile = newFile.replace(/pure/g, "view");
+
+  // Write the new file to disk (temporarily)
+  // Create the temp file by replacing the previous file name with MudGasReport
+  const tempFileName = path.replace(/\.t\.sol$/, "MudGasReport.t.sol");
+  writeFileSync(tempFileName, newFile);
+
+  // Run the generated file using forge
+  const child = execa("forge", ["test", "--match-path", tempFileName, "-vvv"], {
+    stdio: ["inherit", "pipe", "inherit"],
+  });
+
+  // Extrect the logs from the child process
+  let logs = "";
+  try {
+    logs = (await child).stdout;
+    rmSync(tempFileName);
+  } catch (e: any) {
+    console.log(e.stdout ?? e);
+    console.log(chalk.red("\n-----------\nError while running the gas report (see above)"));
+    rmSync(tempFileName);
+    process.exit();
+  }
```

Comment:
With `process.exit()` here, I don't know if this would work, but you might put `rmSync` into `finally`
```ts
try {
 ...
} catch (...) {
  ...
} finally {
  rmSync(tempFileName);
}
```
(unclear if the process exits before it gets here)

Reply:
yeah I had that initially but then changed it in assumption of `process.exit()` terminating the script immediately without getting to `finally` (didn't test this assumption though)